Lib/inspect.py

```diff
@@ -819,7 +819,9 @@ def getfile(object):
     if ismodule(object):
         if getattr(object, '__file__', None):
             return object.__file__
-        raise TypeError('{!r} is a built-in module'.format(object))
+        if object.__spec__ is not None:
+            raise TypeError(f'{object!r} is a built-in module')
+        raise TypeError(f'Custom module: {object!r} cannot get source')
```

Comment:
```suggestion
        if getattr(object, '__spec__', None) is not None:
            raise TypeError(f'{object!r} is a built-in module')
        raise TypeError(f'Cannot get source from {object!r}')
```

Maybe like so? Sorry, I missed it the first time.

Reply:
yes better, but I check seems every guys here have a `__spec__` unless user drop it.
is it needed change to getattr?

Reply:
checked the if getattr(object, '__spec__', None) is not None: one is better

Reply:
seems not right, will roll back